src/Simplify.cpp

```diff
@@ -525,7 +538,11 @@ class Simplify : public IRMutator {
         const Select *select_a = a.as<Select>();
         const Select *select_b = b.as<Select>();
 
-        if (const_int(a, &ia) &&
+        if (is_error_call(call_a)) {
```

Comment:
Yeah, this is turning into an unexpected can of worms. I'm gonna hold off further work on this till Monday; let's discuss it more then.